GitUI/CommitInfo/CommitInfo.cs

```diff
@@ -144,6 +144,23 @@ public CommitInfo()
             rtbxCommitMessage.Height = 1;
         }
 
+        /// <summary>
+        /// Clean up any resources being used.
+        /// </summary>
+        /// <param name="disposing">true if managed resources should be disposed; otherwise, false.</param>
+        protected override void Dispose(bool disposing)
+        {
+            if (disposing)
+            {
+                _asyncLoadCancellation.CancelCurrent();
+                ThreadHelper.JoinPendingOperations(); // those run with FileAndForget
```

Comment:
Is this needed in other forms too?

Reply:
> Is this needed in other forms too?

Yes, this is very likely: to be handled with #7792.